File: backend/app/models/guest_link.py
import uuid
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, Text, func, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database.base import Base


class GuestLink(Base):
    __tablename__ = "guest_links"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()")
    )

    guest_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("guests.id", ondelete="CASCADE"), nullable=False
    )

    label: Mapped[str] = mapped_column(String, nullable=False)
    url: Mapped[str] = mapped_column(Text, nullable=False)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )

    guest: Mapped["Guest"] = relationship(back_populates="guest_links")

File: frontend/services/guests/types/guest-link.ts
/** Mirrors backend/app/schemas/guest_link.py. */
export type GuestLink = {
  id: string;
  guest_id: string;
  label: string;
  url: string;
  created_at: string;
};

export type GuestLinkCreateInput = {
  label: string;
  url: string;
};

export type GuestLinkUpdateInput = Partial<GuestLinkCreateInput>;
